graylog2-server/src/main/java/org/graylog2/cluster/lock/MongoLockService.java

```diff
@@ -98,23 +99,30 @@ private void ensureTTLIndex(MongoCollection<Document> collection, Bson updatedAt
         collection.createIndex(updatedAtKey, indexOptions);
     }
 
-    /**
-     * Request a lock. If a lock already exists, the lock expiry time will be extended.
-     *
-     * @param resource Unique identifier for the resource that should be guarded by this lock.
-     * @return A {@link Lock} object, if a lock was obtained. An empty {@link Optional}, if no lock could be acquired.
-     */
     @Override
-    public Optional<Lock> lock(@Nonnull String resource) {
+    public Optional<Lock> lock(@Nonnull String resource, @Nullable String lockContext) {
+        if (lockContext == null) {
+            return doLock(resource, nodeId.toString());
+        }
+        return doLock(resource, nodeId.toString() + "-" + lockContext);
```

Comment:
Let's use a small helper method to build the `lockedBy` parameter from the node ID and the `lockContext` to avoid duplication. We can also move the `null` check into that method and shorten the `lock` and `unlock` methods.